### PR TITLE
Unit testing for registration!!

### PR DESCRIPTION
**Pull request by lindsayrmaher**
#### What's this PR do?

This PR adds unit tests for our new registration page(s)!! I duplicated the registration testing we already have implemented for various registration use cases using our new registration form, as well as building expected use case tests for the about page and subscriptions page. 

There was a slight code tweak needed in order to text out a multiple-checkbox field that I would like to clean up, but overall these should be at least a start of coverage!

#### How should this be reviewed?

Run the tests to double check that they pass all the specs. Be sure that the logic matches up with the expected behavior for our current process.

#### Relevant Tickets
[169019608](https://www.pivotaltracker.com/story/show/169019608)

#### Checklist
- [ ] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.

### CODE REVIEW

**Comment by mendelB** [comment]
> I duplicated the registration testing we already have implemented for various registration use cases using our new registration form

@lindsayrmaher I wonder if it might make more sense, instead of duplicating the existing tests, to _update_ them to work with the new flow? Since we're going to be flipping the switch to the new flow anyways and retiring the current one. 

So we'd expect some of them to initially fail until we flip the switch and then we'd be in the green? (Maybe we even flip the routing switch in this PR as a last commit to turn the tests green 🤔)

**Comment by lindsayrmaher** [comment]
@mendelB I decided to duplicate rather than update all the tests since we haven't deprecated current registration yet and I felt like we would still want to be testing that until it's turned off. @DFurnes and I talked through it IRL but I didn't make note of it anywhere.

**Comment by lindsayrmaher** [comment]
So potentially it would make more sense to update the registerUpdated helper function to point to the 'register' route and just get rid of the current tests for registration. We could potentially not push the tests to prod until registration has gone live? 🤔

**Comment by mendelB** [comment]
> So potentially it would make more sense to update the registerUpdated helper function to point to the 'register' route and just get rid of the current tests for registration. We could potentially not push the tests to prod until registration has gone live? 🤔

Yeah, I _think_ so!

**Comment by lindsayrmaher** [comment]
I now realize that's basically what you said in your first comment 🤦‍♀  Ok! I'm going to update with just our new registration used

File: resources/views/auth/register-beta.blade.php [hunk 1]
```diff
@@ -45,7 +45,7 @@
         </div>
 
         <div class="form-item -padded">
-            <input type="submit" class="button capitalize" value="Create Account">
+            <input type="submit" id='register-submit' class="button capitalize" value="Create Account">
```
[the remark this answers is not in the document]
Comment (lindsayrmaher) [reply resources/views/auth/register-beta.blade.php]: wait should those id's match?

[the remark this answers is not in the document]
Comment (mendelB) [reply resources/views/auth/register-beta.blade.php]: Yeah I think they need to -- you can just update both to `profile-register-form`

File: tests/TestCase.php [hunk 2]
```diff
@@ -87,4 +87,21 @@ public function register()
...
+
+    /**
+     * Register a new user account with updated registration page.
+     */
+    public function registerUpdated()
+    {
+        // Make sure we're logged out before trying to register.
+        auth('web')->logout();
+
+        $this->visit('register-beta');
```
Comment (mendelB) [inline tests/TestCase.php]: Oh and this should be `/register` now!

File: resources/views/auth/register-beta.blade.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mendelB) [reply resources/views/auth/register-beta.blade.php]: This was all a huge mistake! I thought you'd updated the `form` and not the `input`. Just disregard everything I say till the end of time

**Pull request merged**